packages/react-components/react-skeleton/src/components/Skeleton/Skeleton.types.ts

```diff
@@ -1,17 +1,39 @@
 import type { ComponentProps, ComponentState, Slot } from '@fluentui/react-utilities';
 
 export type SkeletonSlots = {
-  root: Slot<'div'>;
+  /**
+   * The root slot of the `Skeleton` is the container that will contain the slots that make up a `Skeleton`
+   * and any data that the `Skeleton` will load. The default html element is a `div`.
+   */
+  root: NonNullable<Slot<'div'>>;
 };
 
 /**
  * Skeleton Props
  */
-export type SkeletonProps = ComponentProps<SkeletonSlots> & {};
+export type SkeletonProps = Omit<ComponentProps<Partial<SkeletonSlots>>, 'width'> & {
+  /**
+   * Sets the width value of the skeleton wrapper.
+   * @defaultValue 100%
+   */
+  width?: number | string;
+
+  /**
+   * Sets the appearance of the Skeleton.
+   * @defaultValue opaque
+   */
+  appearance?: 'opaque' | 'translucent';
+
+  /**
+   * The animation type for the Skeleton
+   * @defaultValue wave
+   */
+  animation?: 'wave' | 'pulse';
+};
 
 /**
  * State used in rendering Skeleton
  */
-export type SkeletonState = ComponentState<SkeletonSlots>;
-// TODO: Remove semicolon from previous line, uncomment next line, and provide union of props to pick from SkeletonProps.
-// & Required<Pick<SkeletonProps, 'propName'>>
+export type SkeletonState = ComponentState<SkeletonSlots> &
+  Required<Pick<SkeletonProps, 'animation'>> &
+  Pick<SkeletonProps, 'appearance'>;
```

Comment:
I think appearance should be required right? you would have to set the default appearance to opaque.
```suggestion
export type SkeletonState = ComponentState<SkeletonSlots> &
  Required<Pick<SkeletonProps, 'animation' | 'appearance'>>;
```